components/provisioner/internal/provisioning/input_converter.go

```diff
@@ -88,27 +90,65 @@ func (c converter) gardenerConfigFromInput(runtimeID string, input *gqlschema.Ga
 		Name:                                c.createGardenerClusterName(),
 		ProjectName:                         c.gardenerProject,
 		KubernetesVersion:                   input.KubernetesVersion,
-		VolumeSizeGB:                        input.VolumeSizeGb,
-		DiskType:                            input.DiskType,
-		MachineType:                         input.MachineType,
 		Provider:                            input.Provider,
-		Purpose:                             input.Purpose,
-		LicenceType:                         input.LicenceType,
+		Region:                              input.Region,
 		Seed:                                util.UnwrapStr(input.Seed),
 		TargetSecret:                        input.TargetSecret,
+		MachineType:                         input.MachineType,
+		DiskType:                            input.DiskType,
+		VolumeSizeGB:                        input.VolumeSizeGb,
 		WorkerCidr:                          input.WorkerCidr,
-		Region:                              input.Region,
 		AutoScalerMin:                       input.AutoScalerMin,
 		AutoScalerMax:                       input.AutoScalerMax,
 		MaxSurge:                            input.MaxSurge,
 		MaxUnavailable:                      input.MaxUnavailable,
+		Purpose:                             input.Purpose,
+		LicenceType:                         input.LicenceType,
 		EnableKubernetesVersionAutoUpdate:   util.BoolFromPtrOrDefault(input.EnableKubernetesVersionAutoUpdate, c.defaultEnableKubernetesVersionAutoUpdate),
 		EnableMachineImageVersionAutoUpdate: util.BoolFromPtrOrDefault(input.EnableMachineImageVersionAutoUpdate, c.defaultEnableMachineImageVersionAutoUpdate),
 		ClusterID:                           runtimeID,
 		GardenerProviderConfig:              providerSpecificConfig,
 	}, nil
 }
 
+func (c converter) UpgradeShootInputToGardenerConfig(input gqlschema.GardenerUpgradeInput, config model.GardenerConfig) (model.GardenerConfig, apperrors.AppError) {
+
+	providerSpecificConfig, err := c.providerSpecificConfigFromInput(input.ProviderSpecificConfig)
+	if err != nil {
+		providerSpecificConfig = config.GardenerProviderConfig
+	}
```

Comment:
I do not like the way we handle it. In case of an error, we will just fail silently and use the old config.